clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/PlaintextConsumerFetchTest.java

```diff
@@ -0,0 +1,486 @@
+        }
+    }
+
+    @ClusterTest
+    public void testClassicConsumerFetchHonoursFetchSizeIfLargeRecordNotFirst() throws ExecutionException, InterruptedException {
+        testFetchHonoursFetchSizeIfLargeRecordNotFirst(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerFetchHonoursFetchSizeIfLargeRecordNotFirst() throws ExecutionException, InterruptedException {
+        testFetchHonoursFetchSizeIfLargeRecordNotFirst(GroupProtocol.CONSUMER);
+    }
+
+    private void testFetchHonoursFetchSizeIfLargeRecordNotFirst(GroupProtocol groupProtocol) throws ExecutionException, InterruptedException {
+        int maxFetchBytes = 10 * 1024;
+        checkFetchHonoursSizeIfLargeRecordNotFirst(Map.of(
+            GROUP_PROTOCOL_CONFIG, groupProtocol.name().toLowerCase(Locale.ROOT),
+            FETCH_MAX_BYTES_CONFIG, maxFetchBytes
+        ), maxFetchBytes);
+    }
+
+    @ClusterTest
+    public void testClassicConsumerFetchHonoursMaxPartitionFetchBytesIfLargeRecordNotFirst() throws ExecutionException, InterruptedException {
+        testFetchHonoursMaxPartitionFetchBytesIfLargeRecordNotFirst(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerFetchHonoursMaxPartitionFetchBytesIfLargeRecordNotFirst() throws ExecutionException, InterruptedException {
+        testFetchHonoursMaxPartitionFetchBytesIfLargeRecordNotFirst(GroupProtocol.CONSUMER);
+    }
+
+    private void testFetchHonoursMaxPartitionFetchBytesIfLargeRecordNotFirst(GroupProtocol groupProtocol) throws ExecutionException, InterruptedException {
+        int maxFetchBytes = 10 * 1024;
+        checkFetchHonoursSizeIfLargeRecordNotFirst(Map.of(
+            GROUP_PROTOCOL_CONFIG, groupProtocol.name().toLowerCase(Locale.ROOT),
+            MAX_PARTITION_FETCH_BYTES_CONFIG, maxFetchBytes
+        ), maxFetchBytes);
+    }
+
+    private void checkFetchHonoursSizeIfLargeRecordNotFirst(
+        Map<String, Object> config, 
+        int largeProducerRecordSize
+    ) throws ExecutionException, InterruptedException {
+        try (Consumer<byte[], byte[]> consumer = cluster.consumer(config);
+             Producer<byte[], byte[]> producer = cluster.producer()
+        ) {
+            var smallRecord = new ProducerRecord<>(
+                tp.topic(),
+                tp.partition(),
+                "small".getBytes(),
+                "value".getBytes()
+            );
+            var largeRecord = new ProducerRecord<>(
+                tp.topic(),
+                tp.partition(),
+                "large".getBytes(),
+                new byte[largeProducerRecordSize]
+            );
+
+            producer.send(smallRecord).get();
+            producer.send(largeRecord).get();
+
+            // we should only get the small record in the first `poll`
+            consumer.assign(List.of(tp));
+            var records = consumer.poll(Duration.ofMillis(20000));
+            assertEquals(1, records.count());
```

Comment:
How about reusing `consumeRecords`?

```suggestion
            var records = consumeRecords(consumer, 1);
            assertEquals(1, records.size());
```